lib/model/unittest/CMetricModelTest.cc

```diff
@@ -600,7 +600,7 @@ BOOST_FIXTURE_TEST_CASE(testProbabilityCalculationForMedian, CTestFixture) {
         if (i == anomalousBucket) {
             values.push_back(0.0);
             values.push_back(mean * 3.0);
-            values.push_back(mean * 3.0);
+            values.push_back(mean * 6.0);
```

Comment:
Why did this have to change?  Surely the median of 0, 3 and 3 is still 3?  If so it might be nice for the test to alternate the final value between 3 and 6 for consecutive buckets to prove that 0, 3, 3 and 0, 3, 6 both have median 3.

Reply:
This is just an artefact of how duplicate values are handled. Previously, we used piecewise constant which either uses the centre if "cumulative count" / "count" first exceeds the quantile at a centre or the mid point between two centres otherwise. Linear interpolation assumes values are uniformly distributed on the interval which extends from [1.5, 3] so it uses (1.5 + 0.25) * mean in the case that you have {0, 3, 3}. If I just present you with the buckets (0, 1) and (3, 2) they could just as well have generated from say {0, 2.5, 3.5} etc. So this is just the nature of approximation: sometimes we get better results one way sometimes another. If you look at CQuantileSketchTest you'll see for random data we get around 1/5 the error from linear interpolation. That said, I probably can be smarter here. The crux is whether we can work out that we have't actually merged any buckets based on class invariants. I'll have a think if I can do something better around this.

Reply:
Ok it turns out it is fairly easy to deduce whether we have merged values in nearly all cases and use exact interpolation in this case. See [a8ad541](https://github.com/elastic/ml-cpp/pull/2367/commits/a8ad54124454a056dc93d118eafc2c73fc1c2b61).